handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -673,6 +673,204 @@ static ApplicationProtocolConfig toApplicationProtocolConfig(Iterable<String> ne
         return apn;
     }
 
+    /**
+     * Creates a builder for new client-side {@link SslContext}.
+     */
+    public static Builder newBuilderForClient() {
+        return new Builder(true);
+    }
+
+    /**
+     * Creates a builder for new server-side {@link SslContext}.
+     *
+     * @param keyCertChainFile an X.509 certificate chain file in PEM format
+     * @param keyFile a PKCS#8 private key file in PEM format
+     */
+    public static Builder newBuilderForServer(File keyCertChainFile, File keyFile) {
+        return new Builder(false).keyManager(keyCertChainFile, keyFile);
+    }
+
+    /**
+     * Creates a builder for new server-side {@link SslContext}.
+     *
+     * @param keyCertChainFile an X.509 certificate chain file in PEM format
+     * @param keyFile a PKCS#8 private key file in PEM format
+     * @param keyPassword the password of the {@code keyFile}, or {@code null} if it's not
+     *     password-protected
+     */
+    public static Builder newBuilderForServer(File keyCertChainFile, File keyFile,
+            String keyPassword) {
+        return new Builder(false).keyManager(keyCertChainFile, keyFile, keyPassword);
+    }
+
+    /** Builder for configuring a new SslContext for creation. */
+    public static final class Builder {
+        private final boolean forClient;
+        private SslProvider provider;
+        private File trustCertChainFile;
+        private TrustManagerFactory trustManagerFactory;
+        private File keyCertChainFile;
+        private File keyFile;
+        private String keyPassword;
+        private KeyManagerFactory keyManagerFactory;
+        private Iterable<String> ciphers;
+        private CipherSuiteFilter cipherFilter = IdentityCipherSuiteFilter.INSTANCE;
+        private ApplicationProtocolConfig apn;
+        private long sessionCacheSize;
+        private long sessionTimeout;
+
+        private Builder(boolean forClient) {
```

Comment:
Increase accessor visibility of this method to avoid synthetic method generation.  Or consider making this public and removing the helper factory methods.